### PR TITLE
feat: log ssh exit error

### PR DESCRIPTION
**Pull request by Daz-3ux**
### Describe what this PR does / why we need it
tell the user that the error reason is ssh session exit meet error, 
not just display the error number.

### Does this pull request fix one issue?
Fixes: #2216

### CODE REVIEW

**Comment by kakaZhou719** [comment]
@Daz-3ux ，we can not modify vendor files directly unless we update go mod. BTW, try to fix it at those code: https://github.com/sealerio/sealer/blob/150c4cbaba9ce0942141c6f30c14d7cf7821521d/utils/ssh/sshcmd.go#L46

**Comment by Daz-3ux** [comment]
> @Daz-3ux ，we can not modify vendor files directly unless we update go mod. BTW, try to fix it at those code:
> 
> https://github.com/sealerio/sealer/blob/150c4cbaba9ce0942141c6f30c14d7cf7821521d/utils/ssh/sshcmd.go#L46

Of course, I will fix it

**Comment by Daz-3ux** [comment]
@kakaZhou719 
I commit a new, but now stderr is being passed twice. Please check if it's feasible to log the error in this way

**Comment by Daz-3ux** [comment]
> 

It can print error message now and not just error code
image

File: utils/ssh/sshcmd.go [hunk 1]
```diff
@@ -100,11 +100,11 @@ func (s *SSH) CmdAsync(host net.IP, hostEnv map[string]string, cmds ...string) e
 				return fmt.Errorf("failed to start command %s on %s: %v", cmd, host, err)
 			}
 
-			ReadPipe(stdout, stderr, s.AlsoToStdout)
+			errMsg := ReadPipe(stdout, stderr, s.AlsoToStdout)
```
Comment (kakaZhou719) [inline utils/ssh/sshcmd.go:103]: how do we read message from a chan ? i think it is no need to return it, just print it to stdout

File: utils/ssh/utils.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ func fileExist(path string) bool {
 	return err == nil || os.IsExist(err)
 }
 
-func ReadPipe(stdout, stderr io.Reader, alsoToStdout bool) {
+func ReadPipe(stdout, stderr io.Reader, alsoToStdout bool) error {
 	var combineSlice []string
```
Comment (kakaZhou719) [inline utils/ssh/utils.go:63]: combineSlice will contains all the cmd output. so we do new combineErrorSlice to gather  error message .

File: utils/ssh/sshcmd.go (hunk 1 above)
Comment (Daz-3ux) [reply utils/ssh/sshcmd.go:103]: thx for your review, I will fix my code

File: utils/ssh/utils.go [hunk 3]
```diff
@@ -60,18 +60,21 @@ func fileExist(path string) bool {
 }
 
 func ReadPipe(stdout, stderr io.Reader, alsoToStdout bool) {
-	var combineSlice []string
-	var combineLock sync.Mutex
+	var combineDoneSlice []string
+	var combineDoneLock sync.Mutex
+	var combineErrSlice []string
```
Comment (kakaZhou719) [inline utils/ssh/utils.go:65]: good , if we want to see specific error, `combineErrSlice` will contain all messages.

Comment (Daz-3ux) [reply utils/ssh/utils.go:65]: Maybe we can pass a flag and filter to readPipe(), then we can select error
such as:
```
if strings.Contains(message, errorFilter) {
			fmt.Println("Specific error:", message)
		}
```

Comment (kakaZhou719) [reply utils/ssh/utils.go:65]: yes, agree with you.@Daz-3ux

**Pull request closed without merging**